### PR TITLE
Diff-based config file composition

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: prestodb/centos6-oj8/files/usr/local/bin/apply-diffs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
```
Comment (arhimondr) [inline prestodb/centos6-oj8/files/usr/local/bin/apply-diffs]: This file is unused

Comment (findepi) [reply prestodb/centos6-oj8/files/usr/local/bin/apply-diffs]: in this commit -- yes. It's used in Dockerfiles of images that depend on this image

File: prestodb/centos6-oj8/Dockerfile [hunk 2]
```diff
@@ -16,6 +16,8 @@ MAINTAINER Presto community <https://prestodb.io/community.html>
 ARG JDK_URL
 ARG JDK_PATH
 
+COPY ./files /
```
[the remark this answers is not in the document]
Comment (findepi) [reply prestodb/centos6-oj8/Dockerfile]: in the worst case, a derived image will call `apply-diffs` and fail... but i will have a second look

**Comment by findepi** [comment]
Travis was failing on IOP container build (i didn't change anything for IOP)

**Comment by findepi** [comment]
I've verified the generated config files are identical as original (whitespace-ignoring) with the following:
```
for i in cdh5.13-hive-kerberized hdp2.6-hive-kerberized; do
    for f in $(git diff master.. --numstat | cut -f3- | grep $i/files/ | grep -v apply.diff | sed "s@prestodb/$i/files@@"); do
        echo "$i - $f"
        # using https://github.com/findepi/tooltrash/blob/master/docker-copy-from-image
        diff -bBU2 <(docker-copy-from-image prestodb/$i:6 $f) <(docker-copy-from-image prestodb/$i:latest $f)
    done
done
```

**Comment by findepi** [comment]
@kokosing @arhimondr after third and fourth consideration i think those diff files will be pain to edit, causing more problem that being solved.
Either we find some more usable solution (XML-based, so that it's possible to copy paste required configuration from documentation; maybe XSLT) or stay with those copies, as this seems to be lesser pain.
Thank you for your review and sorry for wasting your time.

**Comment by findepi** [comment]
cc @sopel39

**Pull request closed without merging**